java.lang.NoClassDefFoundError: javax/inject/Provider with Micronaut 3.0.0-SNAPSHOT and GraphQL example

I've migrated micronaut-graphql to Micronaut 3.0.0-SNAPSHOT and then I tried to run one of the examples included in the repository.
The application fails to start with the following error:

```
11:56:34.631 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Failed to load a service: null
java.lang.RuntimeException: Failed to load a service: null
        at io.micronaut.core.io.service.SoftServiceLoader$ServiceInstanceLoader.collect(SoftServiceLoader.java:409)
        at io.micronaut.core.io.service.SoftServiceLoader$UrlServicesLoader.collect(SoftServiceLoader.java:367)
        at io.micronaut.core.io.service.SoftServiceLoader$ServicesLoader.collect(SoftServiceLoader.java:310)
        at io.micronaut.core.io.service.SoftServiceLoader.collectAll(SoftServiceLoader.java:146)
        at io.micronaut.context.DefaultBeanContext.resolveBeanDefinitionReferences(DefaultBeanContext.java:1761)
        at io.micronaut.context.DefaultApplicationContext.resolveBeanDefinitionReferences(DefaultApplicationContext.java:135)
        at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3365)
        at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:245)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:181)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:71)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:311)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
        at example.Application.main(Application.java:26)
Caused by: java.lang.reflect.InvocationTargetException: null
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at io.micronaut.core.io.service.SoftServiceLoader$ServiceInstanceLoader.compute(SoftServiceLoader.java:396)
        at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
Caused by: java.lang.NoClassDefFoundError: javax/inject/Provider
        at io.micronaut.inject.provider.JavaxProviderBeanDefinition.getBeanType(JavaxProviderBeanDefinition.java:44)
        at io.micronaut.inject.provider.AbstractProviderDefinition.<init>(AbstractProviderDefinition.java:62)
        at io.micronaut.inject.provider.JavaxProviderBeanDefinition.<init>(JavaxProviderBeanDefinition.java:34)
        ... 10 common frames omitted
Caused by: java.lang.ClassNotFoundException: javax.inject.Provider
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        ... 13 common frames omitted
```


I'm not 100% sure if this is a bug in core or something missing during the upgrade but I haven't found which dependency triggers this error (if any). In other modules the culprit were `micronaut-cache` or `micronaut-micrometer` because those modules still use `javax.inject.Provider` but in graphql those modules are not used.

The workaround is add `javax.inject` dependency but we need to find out the root cause and fix it. 

Maybe the error is that `JavaxProviderBeanDefinition` is loaded and as it extends from `AbstractProviderDefinition<Provider<Object>>` (Provider here is a javax one), fails.

### Steps to Reproduce

- Checkout the branch in this PR: https://github.com/micronaut-projects/micronaut-graphql/pull/197
- `./gradlew graphql-example-chat:run`


### Environment Information

- **Operating System**: Linux Mint 20.1
- **Micronaut Version:** `3.0.0-SNAPSHOT`
- **JDK Version:** 8

--- a/inject/src/main/java/io/micronaut/inject/provider/AbstractProviderDefinition.java
+++ b/inject/src/main/java/io/micronaut/inject/provider/AbstractProviderDefinition.java
@@ -59,7 +59,11 @@ public abstract class AbstractProviderDefinition<T> implements BeanDefinition<T>
                 Collections.emptyMap()
         );
         metadata.addDeclaredAnnotation(BootstrapContextCompatible.class.getName(), Collections.emptyMap());
-        metadata.addDeclaredAnnotation(Indexes.class.getName(), Collections.singletonMap(AnnotationMetadata.VALUE_MEMBER, getBeanType()));
+        try {
+            metadata.addDeclaredAnnotation(Indexes.class.getName(), Collections.singletonMap(AnnotationMetadata.VALUE_MEMBER, getBeanType()));
+        } catch (NoClassDefFoundError e) {
+            // ignore, might happen if javax.inject is not the classpath
+        }
         annotationMetadata = metadata;
     }
 

